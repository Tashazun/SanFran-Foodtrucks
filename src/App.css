#App-container {
    height: 200vh;
    display: flex;
    flex-direction: column;
}

#App-header {
    height: 75vh;
    background: pink;
}

#App-map-frame {
    height: 100vh;
    background: blue;
}

#App-footer {
    height: 25vh;
    background: #292929;
}
