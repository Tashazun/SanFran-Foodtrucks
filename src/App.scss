#App-container {
    height: 200vh;
    width: 100vw;
    display: flex;
    flex-direction: column;

    #App-header {
        height: 75vh;
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        background: #181e2c;

        #App-hero {
            height: 50%;
            width: 100%;
            text-align: center;
            text-shadow: 0px 0px 1px rgba(0,0,0,0.5);
            background: linear-gradient(#083D3290, #083D3290 ), url('./assets/tobias-zils.jpg') no-repeat center center;
            background-size: cover;
            clip-path: polygon(0 0, 1600px 0, 1600px 87%, 0 100%);

            h2 {
                font-size: 3em;
                color: white;
                font-family: 'Neuton', serif;
            }
        }

        #App-animations {
            display: flex;
            justify-content: space-around;

            h3 {
                color: #369381;
                font-family: 'Sansita', sans-serif;
                font-size: 1.5em;
            }

            img {
                height: 15em;
                width: 13em;
                animation: entry 1s ease-in;
            }

            @keyframes entry {
                0% {
                    opacity: 0;
                    transform: translateY(-.5em);
                }

                100% {
                    opacity: 100%;
                    transform: translateY(0);
                }
            }


            #App-summary {
                display: flex;
                flex-direction: column;
                align-items: center;

                hr {
                    width: 75%;
                }

                h4 {
                    color: white;
                }
            }
        }
    }

    #App-map-frame {
        height: 100vh;
        display: flex;
        align-items: center;
        justify-content: center;
        background: white;
    }

    #App-footer {
        height: 25vh;
        display: flex;
        align-items: flex-end;
        justify-content: space-between;
        background: #181e2c;
        clip-path: polygon(0 100%, 1600px 100%, 1600px 0, 0 20%);

        #App-credits {
            margin: 2%;
            color: #369381;

            p {
                margin: 0;
            }
        }

        #App-links {
            width: 8em;
            margin: 2%;
            display: flex;
            justify-content: space-evenly;
            font-size: 1.5em;

            a {
                color: #369381;
            }
        }
    }
}
