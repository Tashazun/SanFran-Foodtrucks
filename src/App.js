import React, { Component } from 'react';
import MyMap from './MyMap';

import './App.scss';
class App extends Component {

    state = {
        data: null,
    };

    async componentDidMount() {
        try {
            const response = await fetch('https://data.sfgov.org/resource/bbb8-hzi6.json');
            if (!response.ok) {
                throw Error(response.statusText);
            }
            const json = await response.json();
            const newData = await json.map(field => ({
                applicant: field.applicant,
                coords: field.location_2.coordinates,
                endtime: field.endtime,
                location: field.location,
                optionaltext: field.optionaltext,
                starttime: field.starttime,
            }));
            this.setState({ data: newData });
        } catch (err) {
            console.log(err);
        }
    };

    render() {
        const { data } = this.state;

        return (
            <div id='App-container'>
                <header id='App-header'>
                    <div id='App-hero'>
                        <h2>Mobile Eats</h2>
                    </div>
                    <h4>Truck tracker for the San Franciscan foodie.</h4>
                </header>
                <section id='App-map-frame'>
                    <MyMap data={data}/>
                </section>
                <footer id='App-footer'>
                    Photo by Tobias Zils on Unsplash
                </footer>
            </div>
        );
    }
}

export default App;