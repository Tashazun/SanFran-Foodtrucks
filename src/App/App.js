import React, { Component } from 'react';
import MyMap from '../Map/MyMap';

import './App.scss';
import spoon from '../assets/spoon.svg';
import fork from '../assets/fork.svg';
import { FaGithubSquare, FaLinkedinIn, FaTwitter, FaReact } from 'react-icons/fa';

class App extends Component {

    state = {
        data: null,
    };

    async componentDidMount() {
        try {
            const response = await fetch('https://data.sfgov.org/resource/bbb8-hzi6.json');
            if (!response.ok) {
                throw Error(response.statusText);
            }
            const json = await response.json();
            const newData = await json.map(field => ({
                applicant: field.applicant,
                coords: [field.location_2.coordinates[1], field.location_2.coordinates[0]],
                endtime: field.endtime,
                location: field.location,
                optionaltext: field.optionaltext,
                starttime: field.starttime,
            }));
            this.setState({ data: newData });
        } catch (err) {
            console.log(err);
        }
    };

    render() {
        const { data } = this.state;

        return (
            <div id='App-container'>
                <header id='App-header'>
                    <div id='App-hero'>
                        <h2>Mobile Eats</h2>
                    </div>
                    <div id='App-animations'>
                        <img src={fork} alt='animated fork' />
                        <div id='App-summary'>
                            <h3>Truck tracker for the San Franciscan foodie.</h3>
                            <hr />
                            <h4>Simply drag or zoom the map to find your next, perfect meal.</h4>
                        </div>
                        <img src={spoon} alt='animated spoon' />
                    </div>
                </header>
                <section id='App-map-frame'>
                    {data !== null && <MyMap data={data}/>}
                </section>
                <footer id='App-footer'>
                    <div id='App-credits'>
                        <p>Photo by <a href='https://unsplash.com/search/photos/tobias-zils'>Tobias Zils</a> on <a href='https://unsplash.com/'>Unsplash</a></p>
                        <p>Candy coded with <FaReact /> and a few other things.</p>
                    </div>
                    <div id='App-links'>
                        <a href='https://github.com/Tashazun'>
                            <FaGithubSquare />
                        </a>
                        <a href='https://www.linkedin.com/in/tashazuniga/'>
                            <FaLinkedinIn />
                        </a>
                        <a href='https://twitter.com/artofobscura'>
                            <FaTwitter />
                        </a>
                    </div>
                </footer>
            </div>
        );
    }
}

export default App;