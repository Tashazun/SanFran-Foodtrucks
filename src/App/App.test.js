import React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import App from './App';

describe('App', () => {
  it('calls fetch with successful status', () => {
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
    const { container } = render(<App />);

    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith('https://data.sfgov.org/resource/bbb8-hzi6.json');
  })
})