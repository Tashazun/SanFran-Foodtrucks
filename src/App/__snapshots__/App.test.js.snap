// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<div>
  <div
    id="App-container"
  >
    <header
      id="App-header"
    >
      <div
        id="App-hero"
      >
        <h2>
          Mobile 
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 640 512"
            width="1em"
          >
            <path
              d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H112C85.5 0 64 21.5 64 48v48H8c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h272c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H40c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h208c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H8c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h208c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H64v128c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
            />
          </svg>
           Eats
        </h2>
      </div>
      <div
        id="App-animations"
      >
        <img
          alt="animated fork"
          id="animated-fork"
          src="fork.svg"
        />
        <div
          id="App-summary"
        >
          <h3>
            Truck tracker for the San Franciscan foodie.
          </h3>
          <hr />
          <h4>
            Simply drag or zoom the map to find your next, perfect meal.
          </h4>
        </div>
        <img
          alt="animated spoon"
          id="animated-spoon"
          src="spoon.svg"
        />
      </div>
    </header>
    <section
      id="App-map-frame"
    />
    <footer
      id="App-footer"
    >
      <div
        id="App-credits"
      >
        <p>
          Photo by
           
          <a
            href="https://unsplash.com/search/photos/tobias-zils"
          >
            Tobias Zils
          </a>
           
          on 
          <a
            href="https://unsplash.com/"
          >
            Unsplash
          </a>
        </p>
        <p>
          Candy coded with 
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M418.2 177.2c-5.4-1.8-10.8-3.5-16.2-5.1.9-3.7 1.7-7.4 2.5-11.1 12.3-59.6 4.2-107.5-23.1-123.3-26.3-15.1-69.2.6-112.6 38.4-4.3 3.7-8.5 7.6-12.5 11.5-2.7-2.6-5.5-5.2-8.3-7.7-45.5-40.4-91.1-57.4-118.4-41.5-26.2 15.2-34 60.3-23 116.7 1.1 5.6 2.3 11.1 3.7 16.7-6.4 1.8-12.7 3.8-18.6 5.9C38.3 196.2 0 225.4 0 255.6c0 31.2 40.8 62.5 96.3 81.5 4.5 1.5 9 3 13.6 4.3-1.5 6-2.8 11.9-4 18-10.5 55.5-2.3 99.5 23.9 114.6 27 15.6 72.4-.4 116.6-39.1 3.5-3.1 7-6.3 10.5-9.7 4.4 4.3 9 8.4 13.6 12.4 42.8 36.8 85.1 51.7 111.2 36.6 27-15.6 35.8-62.9 24.4-120.5-.9-4.4-1.9-8.9-3-13.5 3.2-.9 6.3-1.9 9.4-2.9 57.7-19.1 99.5-50 99.5-81.7 0-30.3-39.4-59.7-93.8-78.4zM282.9 92.3c37.2-32.4 71.9-45.1 87.7-36 16.9 9.7 23.4 48.9 12.8 100.4-.7 3.4-1.4 6.7-2.3 10-22.2-5-44.7-8.6-67.3-10.6-13-18.6-27.2-36.4-42.6-53.1 3.9-3.7 7.7-7.2 11.7-10.7zm-130 189.1c4.6 8.8 9.3 17.5 14.3 26.1 5.1 8.7 10.3 17.4 15.8 25.9-15.6-1.7-31.1-4.2-46.4-7.5 4.4-14.4 9.9-29.3 16.3-44.5zm0-50.6c-6.3-14.9-11.6-29.5-16-43.6 14.4-3.2 29.7-5.8 45.6-7.8-5.3 8.3-10.5 16.8-15.4 25.4-4.9 8.5-9.7 17.2-14.2 26zm11.4 25.3c6.6-13.8 13.8-27.3 21.4-40.6 7.6-13.3 15.8-26.2 24.4-38.9 15-1.1 30.3-1.7 45.9-1.7 15.6 0 31 .6 45.9 1.7 8.5 12.6 16.6 25.5 24.3 38.7 7.7 13.2 14.9 26.7 21.7 40.4-6.7 13.8-13.9 27.4-21.6 40.8-7.6 13.3-15.7 26.2-24.2 39-14.9 1.1-30.4 1.6-46.1 1.6-15.7 0-30.9-.5-45.6-1.4-8.7-12.7-16.9-25.7-24.6-39-7.7-13.3-14.8-26.8-21.5-40.6zm180.6 51.2c5.1-8.8 9.9-17.7 14.6-26.7 6.4 14.5 12 29.2 16.9 44.3-15.5 3.5-31.2 6.2-47 8 5.4-8.4 10.5-17 15.5-25.6zm14.4-76.5c-4.7-8.8-9.5-17.6-14.5-26.2-4.9-8.5-10-16.9-15.3-25.2 16.1 2 31.5 4.7 45.9 8-4.6 14.8-10 29.2-16.1 43.4zM256.2 118.3c10.5 11.4 20.4 23.4 29.6 35.8-19.8-.9-39.7-.9-59.5 0 9.8-12.9 19.9-24.9 29.9-35.8zM140.2 57c16.8-9.8 54.1 4.2 93.4 39 2.5 2.2 5 4.6 7.6 7-15.5 16.7-29.8 34.5-42.9 53.1-22.6 2-45 5.5-67.2 10.4-1.3-5.1-2.4-10.3-3.5-15.5-9.4-48.4-3.2-84.9 12.6-94zm-24.5 263.6c-4.2-1.2-8.3-2.5-12.4-3.9-21.3-6.7-45.5-17.3-63-31.2-10.1-7-16.9-17.8-18.8-29.9 0-18.3 31.6-41.7 77.2-57.6 5.7-2 11.5-3.8 17.3-5.5 6.8 21.7 15 43 24.5 63.6-9.6 20.9-17.9 42.5-24.8 64.5zm116.6 98c-16.5 15.1-35.6 27.1-56.4 35.3-11.1 5.3-23.9 5.8-35.3 1.3-15.9-9.2-22.5-44.5-13.5-92 1.1-5.6 2.3-11.2 3.7-16.7 22.4 4.8 45 8.1 67.9 9.8 13.2 18.7 27.7 36.6 43.2 53.4-3.2 3.1-6.4 6.1-9.6 8.9zm24.5-24.3c-10.2-11-20.4-23.2-30.3-36.3 9.6.4 19.5.6 29.5.6 10.3 0 20.4-.2 30.4-.7-9.2 12.7-19.1 24.8-29.6 36.4zm130.7 30c-.9 12.2-6.9 23.6-16.5 31.3-15.9 9.2-49.8-2.8-86.4-34.2-4.2-3.6-8.4-7.5-12.7-11.5 15.3-16.9 29.4-34.8 42.2-53.6 22.9-1.9 45.7-5.4 68.2-10.5 1 4.1 1.9 8.2 2.7 12.2 4.9 21.6 5.7 44.1 2.5 66.3zm18.2-107.5c-2.8.9-5.6 1.8-8.5 2.6-7-21.8-15.6-43.1-25.5-63.8 9.6-20.4 17.7-41.4 24.5-62.9 5.2 1.5 10.2 3.1 15 4.7 46.6 16 79.3 39.8 79.3 58 0 19.6-34.9 44.9-84.8 61.4zM256 210.2c25.3 0 45.8 20.5 45.8 45.8 0 25.3-20.5 45.8-45.8 45.8-25.3 0-45.8-20.5-45.8-45.8 0-25.3 20.5-45.8 45.8-45.8"
            />
          </svg>
           and a few other things.
        </p>
      </div>
      <div
        id="App-links"
      >
        <a
          href="https://github.com/Tashazun"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM277.3 415.7c-8.4 1.5-11.5-3.7-11.5-8 0-5.4.2-33 .2-55.3 0-15.6-5.2-25.5-11.3-30.7 37-4.1 76-9.2 76-73.1 0-18.2-6.5-27.3-17.1-39 1.7-4.3 7.4-22-1.7-45-13.9-4.3-45.7 17.9-45.7 17.9-13.2-3.7-27.5-5.6-41.6-5.6-14.1 0-28.4 1.9-41.6 5.6 0 0-31.8-22.2-45.7-17.9-9.1 22.9-3.5 40.6-1.7 45-10.6 11.7-15.6 20.8-15.6 39 0 63.6 37.3 69 74.3 73.1-4.8 4.3-9.1 11.7-10.6 22.3-9.5 4.3-33.8 11.7-48.3-13.9-9.1-15.8-25.5-17.1-25.5-17.1-16.2-.2-1.1 10.2-1.1 10.2 10.8 5 18.4 24.2 18.4 24.2 9.7 29.7 56.1 19.7 56.1 19.7 0 13.9.2 36.5.2 40.6 0 4.3-3 9.5-11.5 8-66-22.1-112.2-84.9-112.2-158.3 0-91.8 70.2-161.5 162-161.5S388 165.6 388 257.4c.1 73.4-44.7 136.3-110.7 158.3zm-98.1-61.1c-1.9.4-3.7-.4-3.9-1.7-.2-1.5 1.1-2.8 3-3.2 1.9-.2 3.7.6 3.9 1.9.3 1.3-1 2.6-3 3zm-9.5-.9c0 1.3-1.5 2.4-3.5 2.4-2.2.2-3.7-.9-3.7-2.4 0-1.3 1.5-2.4 3.5-2.4 1.9-.2 3.7.9 3.7 2.4zm-13.7-1.1c-.4 1.3-2.4 1.9-4.1 1.3-1.9-.4-3.2-1.9-2.8-3.2.4-1.3 2.4-1.9 4.1-1.5 2 .6 3.3 2.1 2.8 3.4zm-12.3-5.4c-.9 1.1-2.8.9-4.3-.6-1.5-1.3-1.9-3.2-.9-4.1.9-1.1 2.8-.9 4.3.6 1.3 1.3 1.8 3.3.9 4.1zm-9.1-9.1c-.9.6-2.6 0-3.7-1.5s-1.1-3.2 0-3.9c1.1-.9 2.8-.2 3.7 1.3 1.1 1.5 1.1 3.3 0 4.1zm-6.5-9.7c-.9.9-2.4.4-3.5-.6-1.1-1.3-1.3-2.8-.4-3.5.9-.9 2.4-.4 3.5.6 1.1 1.3 1.3 2.8.4 3.5zm-6.7-7.4c-.4.9-1.7 1.1-2.8.4-1.3-.6-1.9-1.7-1.5-2.6.4-.6 1.5-.9 2.8-.4 1.3.7 1.9 1.8 1.5 2.6z"
            />
          </svg>
        </a>
        <a
          href="https://www.linkedin.com/in/tashazuniga/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M100.3 480H7.4V180.9h92.9V480zM53.8 140.1C24.1 140.1 0 115.5 0 85.8 0 56.1 24.1 32 53.8 32c29.7 0 53.8 24.1 53.8 53.8 0 29.7-24.1 54.3-53.8 54.3zM448 480h-92.7V334.4c0-34.7-.7-79.2-48.3-79.2-48.3 0-55.7 37.7-55.7 76.7V480h-92.8V180.9h89.1v40.8h1.3c12.4-23.5 42.7-48.3 87.9-48.3 94 0 111.3 61.9 111.3 142.3V480z"
            />
          </svg>
        </a>
        <a
          href="https://twitter.com/artofobscura"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </a>
      </div>
    </footer>
  </div>
</div>
`;
